mac_installer/release_brand.sh

```diff
@@ -0,0 +1,400 @@
+# along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+##
+# Script to create a branded installer such as World Community Grid
+#
+## Note 1: This script currently does not have support for "BOINC+VirtualBox.mpkg"
+## and "BOINC + VirtualBox.mpkg"
+##
+
+## Usage:
+## First a brand description file must be created for the desired brand with the variable below defined
+##  SHORTBRANDNAME="wcgrid"                             ##Used to name the branded files and directories in the BOINC_Installer directory
+##  LONGBRANDNAME="World Community Grid"                ##Used to name the installer package
+##  MANAGERAPPNAME="World Community Grid"               ##The name of the branded manager, replaces BOINC Manager
+##  MANAGERICON="WCGridMgr"                             ##The icon for the branded manager, replaces BOINCMgr.icns
+##  SSAVERAPPNAME="World Community Grid"                ##The branded screen saver name
+##  SSAVERPREVIEW="wcgrid.jpg"                          ##The image used as the preview for the screen saver in system preferences
+##  SSAVERTHUMBNAIL="wcgrid-thumbnail"                  ##The image used as the thumbnail for the screen saver in system preferences
+##  SSAVERLOGO="wcg_ss_logo.jpg"                        ##The image used in the generic screen saver
+##  SKINDIR="World Community Grid"                      ##The branded skin name to use
+##  UNINSTALLERAPPNAME="Uninstall World Community Grid" ##The name of the uninstaller app
+##  UNINSTALLERICON="WCGridUninstaller.icns"            ##The icon for the branded uninstaller
+##  UNINSTALLERTRASHICON="WCGridUninstaller.icns"       ##The icon for uninstaller PutInTrash 
+##  INSTALLERAPPNAME="World Community Grid Installer"   ##The name of the installer app
+##  INSTALLERICON="WCGridInstaller.icns"                ##The icon for the branded installer
+##  READMEFILE="WCGrid-ReadMe.rtf"                      ##The branded readme file
+##  BRANDING_INFO="BrandId=4"                           ##Info to write into the branding file
+##
+## NOTE: This script requires Mac OS 10.6 or later, and uses XCode developer
+##   tools.  So you must have installed XCode Developer Tools on the Mac
+##   before running this script.
+##
+## If you wish to code sign the installer and uninstaller, create a file
+## ~/BOINCCodeSignIdentity.txt whose first line is the code signing identity
+##
+## cd to the boinc directory:
+##
+## Invoke this script with the three parts of version number as arguments and the full path to the brand description file.
+## For example, if the version is 3.2.1:
+##     source [path_to_this_script] 3 2 1 ./mac_installer/WCGridInstaller.environment
+##
+## This will create a directory "BOINC_Installer" in the parent directory of
+## the current directory 
+
+if [ $# -lt 4 ]; then
+echo "Usage:"
+echo "   cd [path]/boinc"
+echo "   source [path_to_this_script] major_version minor_version revision_number brand_description_file"
```

Comment:
This script changes enough environment variables that I _very strongly_ suggest not using `source`. If there is any concern about the script file not having execute permission set you can run the script with `bash script.sh`. Note that `source` is mentioned in the comment above too. Also note that the example line below doesn't use `source` due to lack of space after `.`.